specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json

```diff
@@ -1562,6 +1566,15 @@
       },
       "description": "AgentPoolType represents types of an agent pool."
     },
+    "AgentPoolUpgradeSettings": {
+      "properties": {
+        "MaxSurge": {
+          "type": "string",
```

Comment:
should this be a string or int? what does maxSurge:1 indicate?

Reply:
fixed description. Should be a string because it can also be a percent like kuberentes deployments max surge. 
https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#max-surge